src/common/transformations/include/transformations/utils/print_model.hpp

```diff
@@ -322,8 +323,10 @@ void dump_cpp_style(std::ostream& os, const std::shared_ptr<ov::Model>& model) {
         auto version_info = std::string(type_info.get_version());
         auto type = version_info + "::" + type_info.name;
         auto& rt_info = op->get_rt_info();
-        if (rt_info.count("opset") && rt_info["opset"] == "type_relaxed_opset") {
-            type = std::string("ov::op::TypeRelaxed<") + type + ">";
+        if (std::dynamic_pointer_cast<ov::op::TypeRelaxedBase>(op)) {
```

Comment:
please use ov::as_type_ptr
As far as I know, we have a problem with dynamic_pointer_cast on android 
@t-jankowski

Reply:
Perhaps in the future we'd better configure code analysis to prevent the use of std::dynamic_pointer_cast for ov classes.

Reply:
discussed offline